### ISSUE TITLE
Update to support HS107 dual plug

### ISSUE THREAD

**Issue by zzmac2018** [plasticrake/tplink-smarthome-api#53]
Any chance this could be updated to support the new HS107 dual plugs to enable individual plug control via HomeBridge/Home.app Thanks.

**Comment by DaveGut** [comment]
I am working the HS107 also.  I do not know the command structure yet.  I have captured the system information return.  Would like to know when you solve (I will do same).  
DAve

**Comment by DaveGut** [comment]
Found a link on GitHub with HS300 support.  the on-off command also works for the HS107 (tested and verified).  The data for the plugs is a array called "children" in in System Info.
Link:  p-doyle/Python-KasaSmartPowerStrip

Sample command:  var cmd = '{"context":{"child_ids":["80060FD678276903B4B0F85BB58706221AB494E700"]},"system":{"set_relay_state":{"state":1}}}'

The long string is a single Plug Id (= device Id + 00 or 01).  The remainder is very standard.

I will have my Hubitat driver / Smart App done this week.  SmartThings this weekend.

**Comment by plasticrake** [comment]
Thanks for the link @DaveGut. I actually bought an HS300 today on sale so should be able to have support shortly.

**Comment by timcharper** [comment]
@idcrook buy me one and I'll do it :)

Kidding aside, the easiest way to do this is (IMO) is to set your computer up as a wireless access point. If you have a Mac, then this is pretty easy to do provided you also have an ethernet connection. Then you can connect the plug and your phone to your wireless access point and use wireshark to inspect the traffic. There's a neat wireshark decrypter plugin which will decode the tplink packets for you:

https://raw.githubusercontent.com/softScheck/tplink-smartplug/master/tplink-smarthome.lua

Just open the lua evaluator and evaluate it and presto! The encrypted packets will look like plain JSON.

**Comment by hufftheweevil** [comment]
@plasticrake First off, I've been using your API with great success over the last 2 years, so thank you for your work! I recently bought an HS300, only to realize after plugging everything in that it isn't yet supported here. But then I found this thread and I am excited to see that it may be coming soon! I found the individual plugs, like @DaveGut did, in an array called "children", but there doesn't seem to be a way to control them yet. Looking forward to seeing it added. 👍

**Comment by plasticrake** [comment]
Hi everyone! I added support for HS300 (and I believe HS107, but I don't have one to test). It required a pretty major rewrite of a lot of the code to now include a childId parameter to target the individual outlets. And also rewrite of how discovery works. Now, by default discovery will return a device for each outlet (this can be changed to only return the "parent" device) that will default the childId to that particular outlet, which I think is the best use case for this. So for an HS107 device with outlet named "One" and "Two" discovery will return two plugs with those names and when you turn them on or off it will silently pass the correct childId to the setPowerState function.

I've tested against my HS300 device, but would like to hear from someone with an HS107 device.
